mozregression/cli.py

```diff
@@ -585,8 +595,18 @@ def validate(self):
         options.bits = parse_bits(options.bits or mozinfo.bits)
 
         if options.arch is not None:
-            if options.app not in ("gve", "fenix", "focus"):
-                self.logger.warning("--arch ignored for non Android apps.")
+            if user_defined_bits:
+                self.logger.warning(
+                    "--arch and --bits are passed together. --arch will be preferred."
+                )
+
+            if options.app not in arch_options:
+                self.logger.warning(f"--arch ignored for {options.app}.")
+                options.arch = None
+            elif options.app == "firefox" and mozinfo.os == "mac":
```

Comment:
Should this also apply for `firefox-l10n`?

Reply:
```suggestion
            elif options.app in ("firefox", "firefox-l10n") and mozinfo.os == "mac":
```